service/src/builder.rs

```diff
@@ -291,22 +199,110 @@ fn ensure_binary(package: &Package) -> anyhow::Result<()> {
 
 /// Make sure "cdylib" is set for shuttle-next projects, else set it if possible.
 fn ensure_cdylib(package: &Package) -> anyhow::Result<()> {
-    if package.targets().iter().any(|target| target.is_lib()) {
+    if package.targets.iter().any(is_cdylib) {
         Ok(())
     } else {
         bail!("Your Shuttle next project must be a library. Please add `[lib]` to your Cargo.toml file.")
     }
 }
 
-/// Ensure `panic = "abort"` is not set:
-fn check_no_panic(ws: &Workspace) -> anyhow::Result<()> {
-    if let Some(profiles) = ws.profiles() {
-        for profile in profiles.get_all().values() {
-            if profile.panic.as_deref() == Some("abort") {
-                return Err(anyhow!("Your Shuttle project cannot have panics that abort. Please ensure your Cargo.toml does not contain `panic = \"abort\"` for any profiles."));
-            }
+fn is_cdylib(target: &Target) -> bool {
+    target.kind.iter().any(|kind| kind == "cdylib")
+}
+
+fn compiler(
+    packages: Vec<&Package>,
+    release_mode: bool,
+    wasm: bool,
+    project_path: PathBuf,
+) -> anyhow::Result<Vec<BuiltService>> {
+    let jobs = std::thread::available_parallelism()?.get();
+    let manifest_path = project_path.join("Cargo.toml");
+
+    let mut cargo = std::process::Command::new("cargo");
+
+    cargo
+        .arg("build")
+        .arg("-j")
+        .arg(jobs.to_string())
+        .arg("--manifest-path")
+        .arg(manifest_path);
+
+    for package in packages.clone() {
```

Comment:
You can probably iterate over `&packages` here, and the last loop probably doesn't need a clone.